ktor-client/ktor-client-core/ktor-client-core-js/src/io/ktor/client/engine/js/JsClientEngine.kt

```diff
@@ -0,0 +1,42 @@
+package io.ktor.client.engine.js
+
+import io.ktor.client.call.*
+import io.ktor.client.engine.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import io.ktor.http.content.*
+import io.ktor.util.date.*
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.*
+import kotlinx.coroutines.io.*
+import kotlinx.io.core.*
+import org.khronos.webgl.*
+import org.w3c.fetch.*
+import kotlin.browser.*
+import kotlin.coroutines.*
+
+class JsClientEngine(override val config: HttpClientEngineConfig) : HttpClientEngine {
+    override val dispatcher: CoroutineDispatcher = Dispatchers.Default
+
+    override val coroutineContext: CoroutineContext = dispatcher + SupervisorJob()
+
+    override suspend fun execute(
+        call: HttpClientCall, data: HttpRequestData
+    ): HttpEngineCall = withContext(dispatcher) {
+        val callContext = CompletableDeferred<Unit>(this@JsClientEngine.coroutineContext[Job]) + dispatcher
+
+        val requestTime = GMTDate()
+        val request = DefaultHttpRequest(call, data)
+        val rawResponse = fetch(request.url, request.toRaw())
```

Comment:
I guess that fetch is already pretty compatible.
Would be nice if a later version supports nodejs too.
Maybe two engines (JsBrowserEngine, NodeJsEngine) and this one detecting if running on the browser or nodejs and calling the appropiate subengine?

Reply:
There is an issue with service loader: I can't extract the js engine into a separate dependency.
So we could have only 1 js engine for now. :(

Reply:
Well, maybe an if and couple of methods would do the job. Shouldt be too large.